### PR TITLE
Add support for JSON directory output

### PR DESCRIPTION
**Pull request by denis256**
## Description

Included changes:
* Added `terragrunt-json-out-dir` flag to save plan output in json format
* Updated plan saving to create a directory hierarchy 
* Set file names to fixed name: `tfplan.tfplan`(binary plan)  `tfplan.json` (json plan)
* Documentation update

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

Updated `--terragrunt-out-dir` to use plan from directory hierarchy.
Added new flag `--terragrunt-json-out-dir` to save plan in JSON format.

### CODE REVIEW

File: docs/_docs/02_features/execute-terraform-commands-on-multiple-modules-at-once.md [hunk 1]
```diff
@@ -356,5 +367,41 @@ terragrunt run-all plan -destroy --terragrunt-out-dir /tmp/tfplan
...
+└── project-2
+    └── project-2-app1
+        ├── tfplan.json
+        └── tfplan.tfplan
+```
 Notes:
-* The plan for each module will be saved in a separate file named `tfplan`, with the filename generated based on the module's path.
\ No newline at end of file
+* The plan for each module will be saved the same hierarchy as the module structure.
+* Plan file name is `tfplan.tfplan` for binary plan and `tfplan.json` for JSON plan.
```
Comment (yhakbar) [inline docs/_docs/02_features/execute-terraform-commands-on-multiple-modules-at-once.md:405]:
```suggestion
* The file name for the plan is `tfplan.tfplan` for the plan binary and `tfplan.json` for the plan JSON.
```

File: docs/_docs/04_reference/cli-options.md [hunk 2]
```diff
@@ -1229,4 +1230,13 @@ The list of remote registries to cached by Terragrunt Provider Cache server. By
 **Commands**:
 - [run-all](#run-all)
 
-Specify the plan output directory for the `*-all` commands. Useful to save plan between runs in a single place.
\ No newline at end of file
+Specify the plan output directory for the `*-all` commands. Useful to save plan between runs in a single place.
```
Comment (yhakbar) [inline docs/_docs/04_reference/cli-options.md:1233]:
```suggestion
Specify the plan output directory for the `*-all` commands. Useful to save plans between runs in a single place.
```

File: docs/_docs/04_reference/cli-options.md [hunk 3]
```diff
@@ -1229,4 +1230,13 @@ The list of remote registries to cached by Terragrunt Provider Cache server. By
...
+
+### terragrunt-json-out-dir
+
+**CLI Arg**: `--terragrunt-json-out-dir`
+**Environment Variable**: `TERRAGRUNT_JSON_OUT_DIR`
+**Commands**:
+- [run-all](#run-all)
+
+Specify the output directory for the `*-all` commands to store plan in JSON format. Useful to read plan programmatically.
```
Comment (yhakbar) [inline docs/_docs/04_reference/cli-options.md:1242]:
```suggestion
Specify the output directory for the `*-all` commands to store plans in JSON format. Useful to read plans programmatically.
```

File: options/options.go [hunk 4]
```diff
@@ -300,6 +300,9 @@ type TerragruntOptions struct {
 
 	// Folder to store output files.
 	OutputFolder string
+
+	// Folder to store output files.
```
Comment (yhakbar) [inline options/options.go:304]:
```suggestion
	// Folder to store JSON representation of output files.
```

**Review by yhakbar — APPROVED** [review]

**Review by Resonance1584 — APPROVED** [review]
Looks great!

**Pull request merged**